frontend/src/components/module-editor/content/EditableContentBlock.tsx

```diff
@@ -39,7 +47,27 @@ const EditableContentBlock = ({
   index: number;
 }): React.ReactElement => {
   const [isHovered, setIsHovered] = useState(false);
+  const [modalType, setModalType] = useState(ModalType.EDIT);
   const { isOpen, onOpen, onClose } = useDisclosure();
```

Comment:
Rather than having this modal type, it would be simpler to just keep two states: one for whether the delete modal is visible, and one for whether the edit modal is visible. Something like:

```suggestion
  const { isOpen: isEditModalOpen, onOpen: onEditModalOpen, onClose: onEditModalClose } = useDisclosure();
  const { isOpen: isDeleteModalOpen, onOpen: onDeleteModalOpen, onClose: onDeleteModalClose } = useDisclosure();
```